models/migrations/v1_21/v276.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_21 //nolint
+
+import (
+	"code.gitea.io/gitea/models/migrations/base"
+
+	"xorm.io/xorm"
+)
+
+type BadgeUnique struct {
+	Slug        string `xorm:"pk UNIQUE"`
+	Description string
+	ImageURL    string
+}
+
+func (BadgeUnique) TableName() string {
+	return "badge"
+}
+
+func UseSlugInsteadOfIDForBadges(x *xorm.Engine) error {
+	type Badge struct {
+		Slug string
+	}
+	type UserBadge struct {
+		BadgeSlug string `xorm:"INDEX"`
+	}
+	err := x.Sync(new(Badge), new(UserBadge))
+	if err != nil {
+		return err
+	}
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	// add slug to each badge
```

Comment:
I am not quite happy that there is no way we can ensure that the column actually exists.
In other words, there is no way to make this migration re-migratable.
Well… So be it.

Reply:
Wouldn't the sync above ensure that it exists? And it would return an error if it couldn't add the column.